### ISSUE TITLE
unused_variables lint shouldn't warn for unused parameter to default method in trait

### ISSUE THREAD

**Issue by sfackler** [rust-lang/rust#26487]
```rust
trait Foo {
    fn bar(&self, a: i32) {}
}

fn main() {}
```
```
<anon>:2:19: 2:20 warning: unused variable: `a`, #[warn(unused_variables)] on by default
<anon>:2     fn bar(&self, a: i32) {}
                           ^
```

The arguments to default methods in traits form part of the documentation, but it is not uncommon for a default implementation of a method to be a no-op that uses none of its arguments.

**Comment by sanxiyn** [comment]
I implemented this in #26502, but @brson doesn't seem to think it is a good idea. You may want to comment.

**Comment by nikomatsakis** [comment]
After some discussion in the lang team meeting, our feeling was that we ought to not make this change: it's just a bit too special-cased and subtle, and to be consistent the rules ought to apply to other kinds of public functions as well. Instead, the recommended course is to add an #[allow] declaration to your trait or default method.

**Comment by SoniEx2** [comment]
I'd recommend `let _=x;` personally, it's essentially the Rust equivalent of the C `(void)x;`.

That way you neither use `#[allow]`, which can cause issues, nor `_x`, which produces ugly docs.
